File: js/fibonacci.js
const fibonacci = (num) => {
    let result = 1;	
    let prevResult = 0;	
    if (num === 0){	
      return 0;	
    } else if (num === 1){	
      return 1;	
    }	
    for (let i = 2; i <= num; i++){	
      let tempResult = result;	
      result = result + prevResult;	
      prevResult = tempResult;	
    }	
    return result;
}

module.exports = {fibonacci}

console.log(fibonacci(3))

File: python/fibonacci.py
def fibonacci(num):
  if num == 0:
    return 0
  elif num == 1:
    return 1
  else:
    prevResult = 0
    result = 1
    for i in range(2, num + 1):
      tempResult = result	
      result = result + prevResult	
      prevResult = tempResult
  return result

# print(fibonacci(7))

# def fibonacci(n):
#      if n in {0, 1}:  # Base case
#          return n
#      return fibonacci(n - 1) + fibonacci(n - 2)  # Recursive case